### PR TITLE
Adding support for Azure as well as option to install all clouds to s…

### PR DESCRIPTION
**Pull request by rguthriemsft**
Added support for Azure to installation script.  I have also added an option to install all three clouds called all.

### CODE REVIEW

**Comment by rguthriemsft** [comment]
@spinnaker/google-reviewers and @spinnaker/netflix-reviewers PTAL

File: InstallSpinnaker.sh [hunk 1]
```diff
@@ -50,8 +50,8 @@ fi
 
 function print_usage() {
   cat <<EOF
-usage: $0 [--cloud_provider <aws|google|none|both>]
-    [--aws_region <region>] [--google_region <region>]
+usage: $0 [--cloud_provider <aws|google|azure|none|both|all>]
```
Comment (ewiseblatt) [inline InstallSpinnaker.sh]: we should deprecate "both".
I'm not sure "all" is going to make sense in practice, but I guess would not hurt.

File: InstallSpinnaker.sh [hunk 2]
```diff
@@ -60,7 +60,8 @@ usage: $0 [--cloud_provider <aws|google|none|both>]
 
     If run with no arguments you will be prompted for cloud provider and region
 
-    --cloud_provider <arg>      currently supported are google, amazon and none
+    --cloud_provider <arg>      currently supported are google, azure, amazon and none
```
Comment (ewiseblatt) [inline InstallSpinnaker.sh]: is spacing off here?
dont document both. mark it deprecated below where it is handled.

File: InstallSpinnaker.sh [hunk 3]
```diff
@@ -523,7 +541,7 @@ set_defaults_from_environ
 process_args "$@"
 
 if [ "x$CLOUD_PROVIDER" == "x" ]; then
-  read -e -p "Specify a cloud provider (aws|google|none|both): " -i "$DEFAULT_CLOUD_PROVIDER" CLOUD_PROVIDER
+  read -e -p "Specify a cloud provider (aws|azure|google|none|both|all): " -i "$DEFAULT_CLOUD_PROVIDER" CLOUD_PROVIDER
```
Comment (ewiseblatt) [inline InstallSpinnaker.sh]: remove both

File: InstallSpinnaker.sh [hunk 4]
```diff
@@ -537,15 +555,26 @@ case $CLOUD_PROVIDER in
...
       ;;
+  azure)
+      CLOUD_PROVIDER="azure"
+      set_azure_region
+      ;;
   n|no|none)
       CLOUD_PROVIDER="none"
       ;;
-  both|all)
+  both)
```
Comment (ewiseblatt) [inline InstallSpinnaker.sh]: comment both as deprecated

File: InstallSpinnaker.sh [hunk 5]
```diff
@@ -589,6 +618,11 @@ if [[ "${CLOUD_PROVIDER,,}" == "amazon" || "${CLOUD_PROVIDER,,}" == "google" ||
         write_default_value "AWS_VPC_ID" $AWS_VPC_ID
         write_default_value "AWS_SUBNET_ID" $AWS_SUBNET_ID
         ;;
+    azure)
+	write_default_value "SPINNAKER_AZURE_ENABLED" "true"
```
Comment (ewiseblatt) [inline InstallSpinnaker.sh]: is spacing off?

File: InstallSpinnaker.sh [hunk 6]
```diff
@@ -604,6 +638,17 @@ if [[ "${CLOUD_PROVIDER,,}" == "amazon" || "${CLOUD_PROVIDER,,}" == "google" ||
         write_default_value "AWS_VPC_ID" $AWS_VPC_ID
         write_default_value "AWS_SUBNET_ID" $AWS_SUBNET_ID
         ;;
+    all)
```
Comment (ewiseblatt) [inline InstallSpinnaker.sh]: I'm guessing at this point it would be simpler to have independent checks for each and have "both" set google and aws, "all" set each, and the explicit flags set its corresponding one. Then at the end iterate over each check and set as needed.

File: InstallSpinnaker.sh [hunk 7]
```diff
@@ -537,15 +555,26 @@ case $CLOUD_PROVIDER in
       set_google_region
       set_google_zone
       ;;
+  azure)
```
Comment (ewiseblatt) [inline InstallSpinnaker.sh]: Could you fix these so they are orthogonal? Could you refactor it so the provider defaults to none and you can add each independently? That would make it easy to have arbitrary combinations. Then "both" can go away and be replaced by naming aws and google.
